### ISSUE TITLE
FallbackNTP option is not respected by systemd-timesyncd

### ISSUE THREAD

**Issue by majewsky** [systemd/systemd#5091]
### Submission type

  - [X] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

232

### Used distribution

Arch Linux w/ all updates

### In case of bug report: Expected behaviour you didn't see

timesyncd uses the default value of FallbackNTP (i.e. `0.arch.pool.ntp.org` and so on).

### In case of bug report: Unexpected behaviour you saw

timesyncd should use the value of FallbackNTP supplied by my configuration file

### In case of bug report: Steps to reproduce the problem

My configuration only sets the `FallbackNTP` variable once, but not `NTP`.

```bash
$ ls /usr/lib/systemd/timesyncd.conf.d/*.conf
zsh: no matches found: /usr/lib/systemd/timesyncd.conf.d/*.conf
$ ls /run/systemd/timesyncd.conf.d/*.conf
zsh: no matches found: /run/systemd/timesyncd.conf.d/*.conf
$ ls /etc/systemd/timesyncd*
/etc/systemd/timesyncd.conf

/etc/systemd/timesyncd.conf.d:
custom.conf
$ grep '^[^#]' /etc/systemd/timesyncd.conf
[Time]
$ grep '^[^#]' /etc/systemd/timesyncd.conf.d/custom.conf
[Time]
FallbackNTP=ptbtime1.ptb.de
```

I use `grep '^[^#]'` here to skip a lot of verbose comments in `timesyncd.conf` which is the empty default version installed by systemd. With this configuration, I see:

```
$ sudo systemctl restart systemd-timesyncd
$ sudo journalctl -u systemd-timesyncd -n 2
-- Logs begin at Wed 2016-10-19 19:44:28 CEST, end at Mon 2017-01-16 17:15:43 CET. --
Jan 16 17:07:11 poghril systemd[1]: Started Network Time Synchronization.
Jan 16 17:07:11 poghril systemd-timesyncd[26133]: Synchronized to time server 131.234.137.63:123 (0.arch.pool.ntp.org).
```

**Comment by mbiebl** [comment]
This was also reported downstream at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863038 so I looked into this a bit. I used the same configuration as the bug reporter:
```

[Time]
NTP=
FallbackNTP=0.de.pool.ntp.org 1.de.pool.ntp.org 2.de.pool.ntp.org
```

Running systemd-timesyncd from the command line in debug mode gives me
```
# /lib/systemd/systemd-timesyncd 
Added new server 0.debian.pool.ntp.org.
Added new server 1.debian.pool.ntp.org.
Added new server 2.debian.pool.ntp.org.
Added new server 3.debian.pool.ntp.org.
Added new server 0.de.pool.ntp.org.
Added new server 1.de.pool.ntp.org.
Added new server 2.de.pool.ntp.org.
Added new server 3.de.pool.ntp.org.
systemd-timesyncd running as pid 32124
Selected server 0.debian.pool.ntp.org.
Resolving 0.debian.pool.ntp.org...
Resolved address 85.214.194.162:123 for 0.debian.pool.ntp.org.
Resolved address 129.70.132.36:123 for 0.debian.pool.ntp.org.
Resolved address 62.116.162.126:123 for 0.debian.pool.ntp.org.
Resolved address 144.76.208.2:123 for 0.debian.pool.ntp.org.
Selected address 85.214.194.162:123 of server 0.debian.pool.ntp.org.
Connecting to time server 85.214.194.162:123 (0.debian.pool.ntp.org).
```

As it seems, timesyncd *appends* instead of *replaces* the compiled-in fallback list and as 0.debian.pool.ntp.org is resolvable/reachable, it picks that one.

**Comment by mbiebl** [comment]
I posted a PR at #5999 (it handles the configuration similar to resolved). Please have a look.
